cmd/admin-handlers.go

```diff
@@ -778,25 +831,36 @@ func (adminAPI adminAPIHandlers) HealUploadHandler(w http.ResponseWriter, r *htt
 		return
 	}
 
+	resp := HealUploadResponse{}
+
+	writeSuccessResponseHeadersOnly(w)
+
+	keepAlive := newKeepConnUp(w)
+	keepAlive.StartBackground()
+
 	//We are able to use HealObject for healing an upload since an
 	//ongoing upload has the same backend representation as an
 	//object.  The 'object' corresponding to a given bucket,
 	//object and uploadID is
 	//.minio.sys/multipart/bucket/object/uploadID.
 	numOfflineDisks, numHealedDisks, err := objLayer.HealObject(minioMetaMultipartBucket, uploadObj)
 	if err != nil {
-		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
-		return
-	}
+		resp.Error = getAPIErrorFromCode(toAPIErrorCode(err), r.URL.Path)
 
-	jsonBytes, err := json.Marshal(newHealResult(numHealedDisks, numOfflineDisks))
-	if err != nil {
-		writeErrorResponse(w, toAPIErrorCode(err), r.URL)
-		return
+	} else {
+		jsonBytes, err := json.Marshal(newHealResult(numHealedDisks, numOfflineDisks))
+		if err != nil {
+			resp.Error = getAPIErrorFromCode(toAPIErrorCode(err), r.URL.Path)
+
+		} else {
+			resp.HealResult = jsonBytes
```

Comment:
> IIUC, we are sending json value inside an xml response. Isn't that unconventional and possibly inconvenient for alternate minio admin SDK authors (in the future)?

yes you are right, we probably need to change heal result from json to xml, what do you think ?